src/server/graphql/models/utils.js

```diff
@@ -7,38 +7,26 @@ export const defaultResolveFn = (source, args, {fieldName}) => {
 
 export function resolveForAdmin(source, args, ref) {
   return ref.rootValue &&
-    ref.rootValue.authToken &&
-    ref.rootValue.authToken.isAdmin ? defaultResolveFn.apply(this, [source, args, ref]) : null;
+  ref.rootValue.authToken &&
+  ref.rootValue.authToken.isAdmin ? defaultResolveFn.apply(this, [source, args, ref]) : null;
 }
 
 // Stringify an object to handle multiple errors
 // Wrap it in a new Error type to avoid sending it twice via the originalError field
-export const errorObj = obj =>
-  new Error(JSON.stringify(obj));
-
-// Showing a GraphQL error to the client is ugly
-export const prepareClientError = res => {
-  const {errors, data} = res;
-  if (!errors) {
-    return res;
-  }
-  const error = errors[0].message;
-  if (error && error.indexOf('{"_error"') === -1) {
-    console.log('DEBUG PARSING GraphQL Error:', error);
-    return {data, error: JSON.stringify({_error: 'Server error while querying data'})};
-  }
-  return {data, error};
-};
+export const errorObj = obj => new Error(JSON.stringify(obj));
 
 // if the add & update schemas have different required fields, use this
-export const makeRequired = (fields, requiredFieldNames) => {
-  const newFields = Object.assign({}, fields);
-  requiredFieldNames.forEach(name => {
-    newFields[name] = Object.assign({}, newFields[name], {
-      type: new GraphQLNonNull(newFields[name].type)
-    });
+export const nonnullifyInputThunk = (name, inputThunk, requiredFieldNames) => {
```

Comment:
create an input type that has different `NonNull` requirements. To combat the circular reference, the input is a thunk.